src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/consumer/impl/MainConsumerImpl.java

```diff
@@ -85,7 +89,7 @@ public void readAndProcessRetryTopic(@Payload List<ChipsRestInterfacesSend> mess
             processMessage(groupId, messages.get(i), offsets.get(i), partitions.get(i), failedMessageIds);
         }
 
-        if (!failedMessageIds.isEmpty()) {
+        if (doSendMessages && !failedMessageIds.isEmpty()) {
```

Comment:
should be able to test this with feature flag on and off, can check if slack service gets called.

Reply:
Tried to but I think I need help here as I was left wondering how to populate the message id list - it is a local variable remember the concerns over thread safety.

Reply:
is there any reason that the processMessage (and by extension messageProcessorService.processMessage()
Can't return a list of failed messages (or returns an optional List), instead of passing in an empty list that the processor service modifies?
It's not strictly clear anywhere that the list you pass in is going to be modified

Reply:
ignore most of what i said above, why not just return true or false from process message.
If false add the id to the list.
Would make testing easy as you just mock process message returning false